indigo_api/models/tasks.py

```diff
@@ -183,9 +184,35 @@ def decorate_potential_assignees(cls, tasks, country):
             elif task.state == 'pending_review':
                 task.potential_assignees = [u for u in potential_reviewers if task.assigned_to_id != u.id and
                                             (u.has_perm('indigo_api.close_any_task') or task.submitted_by_user_id != u.id)]
+            else:
+                task.potential_assignees = []
+
+            # move the current user first
+            if task.potential_assignees and current_user and current_user.is_authenticated:
+                task.potential_assignees.sort(key=lambda u: 0 if u.id == current_user.id else 1)
+
+        # mark users that have too many tasks
+        cls.decorate_user_task_limits(set(u for task in tasks for u in task.potential_assignees), tasks)
 
         return tasks
 
+    @classmethod
+    def decorate_user_task_limits(cls, users, tasks):
+        """ Count assigned tasks for these users and decorate users that have exceeded their threshold.
+        """
+        states = set(t.state for t in tasks)
+        counts = Task.objects \
+            .filter(assigned_to__in=users, state__in=states) \
+            .values('assigned_to') \
+            .annotate(tasks=Count(1))
+        counts = {c['assigned_to']: c['tasks'] for c in counts}
```

Comment:
I might not be understanding what `tasks=Count(1)` is doing above. 
From my reading, line 204-207 will produce something like
QuerySet[{'assigned_to': 37, 'tasks': 1}, {'assigned_to': 42, 'tasks': 1}, {'assigned_to': 37, 'tasks': 1}]
where `tasks` will always be `1`? 

and line 208 something like 
{37: 1, 42: 1}
because when `37` is encountered for the second time it'll overwrite it?

Reply:
The `tasks=Count(1)` effectively tells SQL to count every row, grouping by `assigned_to`.

```sql
SELECT
  assigned_to,
  count(1)
FROM xxx
GROUP BY assigned_to;
```

https://learnsql.com/blog/difference-between-count-distinct/

Reply:
`count(1)` counts every row. It looks a bit weird, I know. I thought it was a bug when I first saw it used :)

Reply:
ah thanks! I looked at the docs online and they only give field names in the examples :upside_down_face: